tests/test_jws.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) The Diem Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+from diem import jws
+from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey, Ed25519PublicKey
+from cryptography.exceptions import InvalidSignature
+import pytest
+
+
+HEX_KEY = "ccf2700f8b2d001a8caf80ca2bfc5b7cc71df1f799e02c78bc3f07ea3af79a98"
+KEY = Ed25519PrivateKey.from_private_bytes(bytes.fromhex(HEX_KEY))
+PUBLIC_KEY = KEY.public_key()
+HEADERS = {"keyId": "hello", "alg": "EdDSA"}
+MSG = "msg"
+
+
+def test_encode_decode_message():
+    sig = jws.encode(MSG, KEY.sign)
+    assert (
+        sig
+        == b"eyJhbGciOiAiRWREU0EifQ==.bXNn.4_opZfkm3zTZevTdSIiy8T4LEJiwSY-ygYcL9IYY9xApTXQopaG_ktYMk1xn-UU5np4F3FTiKp7IbSkEOFx_BA=="
```

Comment:
Can we steal from an existing JWT library (or reuse it?)- I ask because this is not to spec: https://tools.ietf.org/html/rfc7519#section-7.2 - i.e `==` padding should be omitted

Reply:
for example. https://github.com/jpadilla/pyjwt/ supports this

Reply:
fixed. python base64 urlsafe keeps padding, just need to remove it.

For `jwcrypto`, the primary reason to not use is it couples the key and jws encoding, basically, you need hold the private key for encoding JWS, which you probably don't have control in the SDK lib.

This jws module provides util functions for constructing message and signing_message, and passing in `sign` function for signing message. It's up to the application to control the private key.

Reply:
If application chooses `jwcrypto`, then they don't need this `jws` module. 
Here mostly trying to provide utils functions in case application does not use `jwcrypto` to hold the private key.

Reply:
that makes sense- my concern is us releasing a `jwt` signing that then ends up not being to spec or incompatible with some other library a vasp is using in some way